internal/pkg/cli/app_show.go

```diff
@@ -0,0 +1,330 @@
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+const (
+	applicationShowProjectNamePrompt     = "Which project's applications would you like to show?"
+	applicationShowProjectNameHelpPrompt = "A project groups all of your applications together."
+	applicationShowAppNamePrompt         = "Which application of %s would you like to show?"
+	applicationShowAppNameHelpPrompt     = "The detail of an application will be shown (e.g., endpoint URL, CPU, Memory)."
+)
+
+// ShowAppOpts contains the fields to collect for showing an application.
+type ShowAppOpts struct {
+	ShouldOutputJSON bool
+
+	proj *archer.Project
+	app  *archer.Application
+	envs []*archer.Environment
+	uris []*describe.WebAppURI
+
+	appName string
+
+	storeSvc   storeReader
+	identifier resourceIdentifier
+
+	w io.Writer
+
+	// cached environments in project
+	environments []*archer.Environment
+	*GlobalOpts
+}
+
+// Ask asks for fields that are required but not passed in.
+func (opts *ShowAppOpts) Ask() error {
+	// get project
+	if opts.ProjectName() == "" {
+		projectName, err := opts.selectProject()
+		if err != nil {
+			return err
+		}
+		opts.projectName = projectName
+	}
+
+	// get application of the project
+	if opts.appName == "" {
+		appName, err := opts.selectApplication()
+		if err != nil {
+			return err
+		}
+		opts.appName = appName
+	}
+
+	return nil
+}
+
+// Execute shows the applications through the prompt.
+func (opts *ShowAppOpts) Execute() error {
+	proj, err := opts.storeSvc.GetProject(opts.ProjectName())
+	if err != nil {
+		return fmt.Errorf("getting project: %w", err)
+	}
+	opts.proj = proj
+
+	if opts.appName != "" {
+		app, err := opts.storeSvc.GetApplication(opts.ProjectName(), opts.appName)
+		if err != nil {
+			return fmt.Errorf("getting application: %w", err)
+		}
+		opts.app = app
+
+		envs, err := opts.storeSvc.ListEnvironments(opts.ProjectName())
+		if err != nil {
+			return fmt.Errorf("listing environments: %w", err)
+		}
+		opts.environments = envs
+
+		uris := make([]*describe.WebAppURI, len(opts.environments))
+		for ind, env := range envs {
+			webAppURI, err := opts.identifier.URI(env.Name)
+			if err != nil {
+				if applicationNotDeployed(errors.Unwrap(err)) {
+					uris[ind] = &describe.WebAppURI{
+						DNSName: "",
+						Path:    "",
+					}
+				} else {
+					return fmt.Errorf("retrieving application URI: %w", err)
+				}
+			} else {
+				uris[ind] = webAppURI
+			}
+		}
+		// only include environments that application deploys in
+		for ind, uri := range uris {
+			if uri.DNSName != "" {
+				opts.envs = append(opts.envs, opts.environments[ind])
+				opts.uris = append(opts.uris, uri)
+			}
+		}
+	} else {
+		opts.app = &archer.Application{}
+		opts.envs = make([]*archer.Environment, 0)
+		opts.uris = make([]*describe.WebAppURI, 0)
+	}
```

Comment:
If the application is not found in the project

Reply:
oh i see, we can probably move this logic to a separate function and use [guard statements](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) so that we reduce nesting:

```go
func (o *opts) retrieveData() {
   if o.AppName == "" {
       // set empty case
       return
   }
   // otherwise do the longer logic
}
```